src/app/shared/components/template/components/navigation-bar/navigation-bar.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from "@angular/core";
+import { TemplateBaseComponent } from "../base";
+import { getParamFromTemplateRow, getTemplateNameFromUrl } from "src/app/shared/utils";
+import { parseValueListItems } from "../../utils";
+import { Router } from "@angular/router";
+
+interface IButton {
+  image: string | null;
+  text: string | null;
+  target_template: string | null;
+  name: string | null;
+}
+
+@Component({
+  selector: "plh-navigation-bar",
+  templateUrl: "./navigation-bar.component.html",
+  styleUrls: ["./navigation-bar.component.scss"],
+})
+export class TmplNavigationBarComponent extends TemplateBaseComponent implements OnInit {
```

Comment:
I think fair enough for now to have a standalone component for this, as you are right - there is enough specific behaviour that might otherwise be hard to capture in display group components (namely applying class based on active link). 

For the most part this looks fine to me, although I expect most of the code will disappear when using routerLink and routerLinkActive in the template component instead as suggested